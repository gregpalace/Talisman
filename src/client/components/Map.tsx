import React from 'react';

const Map = (props: any) => {
  return (
    <div
    style={{
      width: '800px',
      height: '400px',
      backgroundColor: 'black',
      border: '4px solid black',
      margin: '10px auto',
    }}
    />
  );
}


export default Map;