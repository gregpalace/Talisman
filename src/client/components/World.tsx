import React from 'react';

import Map from './Map';
import Player from './Player';

const World = (props: any) => {
  return (
    <div
    style={{
      position: 'relative',
      width: '800px',
      height: '400px',
      margin: '20px auto'
    }}>
      <Map />
      <Player />
    </div>
  );
}

export default World;