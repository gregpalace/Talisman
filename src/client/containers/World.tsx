import React from 'react';
import { connect } from 'react-redux';
// import * as actions from '../actions/actions';

import Map from '../components/Map';
import Player from '../components/Player';
import { tiles } from '../../data/maps/1/map1';
import store from '../../store';

interface IPlayerState {
  readonly hp: number;
  readonly mp: number;
  readonly position: number[];
  readonly inventorySlots: number;
}

const mapStateToProps = (state: IPlayerState) => ({
  position: state.position,
});

const World = (props: any) => {
  store.dispatch({
    type: 'ADD_TILES',
    payload: {
      tiles,
    },
  });
  return (
    <div
      style={{
        position: 'relative',
        width: '1000px',
        height: '500px',
        margin: '20px auto',
      }}
    >
      <Map />
      <Player />
    </div>
  );
};

export default connect(mapStateToProps)(World);
