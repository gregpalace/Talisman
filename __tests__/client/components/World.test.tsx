import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Player from '../../../src/client/components/Player';
import Map from '../../../src/client/components/Map';
import World from '../../../src/client/components/Map';

describe('React unit tests', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallow(<World />);
  });

  describe('<World />', () => {
    it('Should render correctly', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render a wrapper with styling properties', () => {
      expect(wrapper.find('style')).toBeTruthy();
      expect(wrapper.prop('style').width).toEqual('800px');
      expect(wrapper.prop('style').height).toEqual('400px');
    });

    it('should render <Map /> and <Player /> components', () => {
      expect(wrapper.find(Map)).toBeDefined();
      expect(wrapper.find(Player)).toBeDefined();
    });
  });
});
