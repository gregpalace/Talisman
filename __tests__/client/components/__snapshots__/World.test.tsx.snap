// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React unit tests <World /> Should render correctly 1`] = `
<div
  style={
    Object {
      "height": "400px",
      "margin": "20px auto",
      "position": "relative",
      "width": "800px",
    }
  }
>
  <Map />
</div>
`;
